test/extended/apiserver/tls.go

```diff
@@ -1,31 +1,146 @@
+	o "github.com/onsi/gomega"
+	"github.com/openshift/library-go/test/library"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 
+	operatorv1client "github.com/openshift/client-go/operator/clientset/versioned"
 	"github.com/openshift/library-go/pkg/crypto"
 	exutil "github.com/openshift/origin/test/extended/util"
 )
 
+const tlsTestDuration = 45 * time.Minute
+const tlsWaitForCleanupDuration = 10 * time.Minute
+
+var _ = g.Describe("[sig-api-machinery][Feature:APIServer][Serial][Slow]", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLIWithoutNamespace("apiserver")
+
+	g.It("TestTLSModernProfile", func() {
+		ctx, ctxCancelFn := context.WithTimeout(context.Background(), tlsTestDuration)
+		defer ctxCancelFn()
+
+		t := g.GinkgoT()
+
+		configClient := oc.AdminConfigClient()
+
+		operatorClient := oc.AdminOperatorClient()
+
+		defer func() {
+			g.By("Cleanup - removing TLS profile")
+
+			cleanupCtx, cancelCleanup := context.WithTimeout(context.Background(), tlsWaitForCleanupDuration)
+			defer cancelCleanup()
+
+			kasStatus, err := operatorClient.OperatorV1().KubeAPIServers().Get(cleanupCtx, "cluster", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			currentRevision := kasStatus.Status.LatestAvailableRevision
+
+			removeModernProfilePatch := `[{"op":"remove","path":"/spec/tlsSecurityProfile"}]`
+
+			_, err = configClient.ConfigV1().APIServers().Patch(cleanupCtx, "cluster", types.JSONPatchType, []byte(removeModernProfilePatch), metav1.PatchOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			_, err = waitForKASIncrementedRevision(cleanupCtx, operatorClient, "cluster", currentRevision)
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}()
+
+		g.By("Checking if TLS 1.2 is usable before the modern TLS profile is applied")
+
+		// We're going to be dialing TCP directly, not connecting over HTTP as usual, so we don't want the protocol on the host.
+		tlsHost := strings.TrimPrefix(oc.AdminConfig().Host, "https://")
+
+		config := &tls.Config{MinVersion: tls.VersionTLS12, MaxVersion: tls.VersionTLS12, InsecureSkipVerify: true}
+
+		conn, err := tls.Dial("tcp4", tlsHost, config)
+		if err != nil {
+			t.Fatalf("Expected success with TLS 1.2 using default profile, got %v", err)
+		} else {
+			t.Log("TLS 1.2 is usable")
+		}
+
+		conn.Close()
+
+		g.By("Applying a JSON Patch to use the modern TLS profile")
+
+		kasStatus, err := operatorClient.OperatorV1().KubeAPIServers().Get(ctx, "cluster", metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		currentKASRevision := kasStatus.Status.LatestAvailableRevision
+
+		addModernProfilePatch := `[{"op":"add","path":"/spec/tlsSecurityProfile","value":{"type":"Modern","modern":{}}}]`
+
+		_, err = configClient.ConfigV1().APIServers().Patch(ctx, "cluster", types.JSONPatchType, []byte(addModernProfilePatch), metav1.PatchOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("Waiting for etcd to stabilize")
```

Comment:
I think we should only wait for `WaitForAPIServerToStabilizeOnTheSameRevision`, as it takes the longest to roll out the servers. Besides, we're only testing the servers, even though updating the profiles affects other components as well.